Plugins/Mute/DataManager.cs

```diff
@@ -1,38 +1,119 @@
-﻿using Data.Models.Client;
+﻿using SharedLibraryCore;
+using SharedLibraryCore.Database.Models;
 using SharedLibraryCore.Interfaces;
 using static System.Enum;
 
 namespace Mute;
 
 public class DataManager
 {
-    public DataManager(IMetaServiceV2 metaService)
+    private readonly IMetaServiceV2 _metaService;
+    private readonly ITranslationLookup _translationLookup;
+
+    public DataManager(IMetaServiceV2 metaService, ITranslationLookup translationLookup)
     {
         _metaService = metaService;
+        _translationLookup = translationLookup;
     }
 
-    private readonly IMetaServiceV2 _metaService;
-
-    public async Task<MuteState> ReadPersistentData(EFClient client)
+    public async Task<MuteStateMeta?> ReadPersistentData(EFClient client)
     {
-        var clientMuteState = client.GetAdditionalProperty<MuteState?>(Plugin.MuteKey) ??
-                              Parse<MuteState>((await _metaService.GetPersistentMeta(Plugin.MuteKey, client.ClientId))?
-                                  .Value ?? nameof(MuteState.Unmuted));
+        // Get and return meta.
+        var clientMuteStateMeta = client.GetAdditionalProperty<MuteStateMeta>(Plugin.MuteKey);
+        if (clientMuteStateMeta is not null) return clientMuteStateMeta;
+
+        // Get and return meta from database.
+        clientMuteStateMeta = await _metaService.GetPersistentMetaValue<MuteStateMeta>(Plugin.MuteKey,
+            client.ClientId);
+
+        if (clientMuteStateMeta is not null)
+        {
+            client.SetAdditionalProperty(Plugin.MuteKey, clientMuteStateMeta);
```

Comment:
So I don't think the data manager should be responsible for setting the additional property here. Thing of this as retrieval only. The manager should be the place to set the additional property when needed